### ISSUE TITLE
Track expense-In self DM, create expense is shown instead track expense

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#52897]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.65-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Isuue was found when executing this PR:** https://github.com/Expensify/App/pull/52786
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Login with new gmail account
3. Open self DM
4. Tap plus icon next to chat input

## Expected Result:
In self DM, track expense must be shown .

## Actual Result:
In self DM, create expense is shown instead track expense.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/22563c18-62fc-447d-a2f4-0b190bcdb3c5

  </details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In self DM, create expense is shown instead track expense.

### What is the root cause of that problem?
We display `Create expense` if `canUseCombinedTrackSubmit` is `true`. 

https://github.com/Expensify/App/blob/461781621a22903b64534202c95811890bf05df4/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L162-L163
### What changes do you think we should make in order to solve the problem?
I think we should display `Track expense` for all case in the plus icon of SelfDM

```
icon:  getIconForAction(CONST.IOU.TYPE.TRACK),
ext: translate('iou.trackExpense'),
```
https://github.com/Expensify/App/blob/461781621a22903b64534202c95811890bf05df4/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L162-L163

### What alternative solutions did you explore? (Optional)
NA

**Comment by abekkala** [comment]
confirmed in slack that it should be track

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Track expense-In self DM, create expense is shown instead track expense

### What is the root cause of that problem?
We set the menu item to create expense if `canUseCombinedTrackSubmit` permission is on for `IOU.TYPE.TRACK` here
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L161-L163
and then we remove the duplicate create expense menu item here because for canUseCombinedTrackSubmit both track and submit are set to Create expense
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L177-L180

and then we set the IOUStartPage title to create expense to similarly for `canUseCombinedTrackSubmit` case here
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/iou/request/IOURequestStartPage.tsx#L58

### What changes do you think we should make in order to solve the problem?

We need to set it to track for `CONST.IOU.TYPE.TRACK` whether or not canUseCombinedTrackSubmit is on
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L161-L163

```
[CONST.IOU.TYPE.TRACK]: {
                icon: getIconForAction(CONST.IOU.TYPE.TRACK),
                text: translate('iou.trackExpense'),
                onSelected: () => selectOption(() => IOU.startMoneyRequest(CONST.IOU.TYPE.TRACK, report?.reportID ?? '-1'), true),
            },
```
and update IOUStartPage title here
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/iou/request/IOURequestStartPage.tsx#L58

```
        [CONST.IOU.TYPE.TRACK]: translate('iou.trackExpense'),

```
then we also don't need the filter code here anymore
https://github.com/Expensify/App/blob/70461d0871b9db9f149437e923c695476b02ae84/src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx#L177-L180
```
        return moneyRequestOptionsList;

```
### What alternative solutions did you explore? (Optional)

**Comment by ahmedGaber93** [comment]
reviewing today

**Comment by ahmedGaber93** [comment]
https://github.com/Expensify/App/blob/10a7be853c9e449f5c680cedaeb298dfae95afae/src/libs/Permissions.ts#L32-L36

According to the above comment, this is A/B testing feature, and "create expense" will display for even number account IDs. So I think this intended for A/B testing purpose and not an issue

**Comment by FitseTLT** [comment]
> https://github.com/Expensify/App/blob/10a7be853c9e449f5c680cedaeb298dfae95afae/src/libs/Permissions.ts#L32-L36
> 
> According to the above comment, this is A/B testing feature, and "create expense" will display for even number account IDs. So I think this intended for A/B testing purpose and not an issue

Yeah but we only want that to replace submit and not for track even for even numbered account ids that the purpose of this issue and also there is a slack confirmation [here](https://github.com/Expensify/App/issues/52897#issuecomment-2502021737)
I think it is better to have an engineering :eyes:

**Comment by FitseTLT** [comment]
> https://github.com/Expensify/App/blob/10a7be853c9e449f5c680cedaeb298dfae95afae/src/libs/Permissions.ts#L32-L36
> 
> According to the above comment, this is A/B testing feature, and "create expense" will display for even number account IDs. So I think this intended for A/B testing purpose and not an issue

This is not beta feature and it affects users in production with even numbered account ids and the problem here is showing `create` in place of `track` we only intended to replace `submit` with `create` so this is still a valid bug that affects users in production hence the slack confirmation [here](https://github.com/Expensify/App/issues/52897#issuecomment-2502021737)

cc @abekkala

**Comment by ahmedGaber93** [comment]
> we only intended to replace submit with create 

@FitseTLT The https://github.com/Expensify/App/issues/49691 OP addressed the three cases that needed to replace with `create` and `track` is one of them. See the Goal section.

**Comment by ahmedGaber93** [comment]
> confirmed in slack that it should be track

@abekkala Just for confirmation, was it taken into consideration that this behavior is not a bug and that it is for A/B testing purpose, and you decide to disable A/B testing here?

More information here https://github.com/Expensify/App/issues/49691 in OP Goal section.

**Comment by ahmedGaber93** [comment]
@abekkala bump https://github.com/Expensify/App/issues/52897#issuecomment-2512347136

**Comment by abekkala** [comment]
@grgia since you're familiar with this can you confirm https://github.com/Expensify/App/issues/52897#issuecomment-2512439651

**Comment by ahmedGaber93** [comment]
WDYT @grgia ?

**Comment by ahmedGaber93** [comment]
Bumped @grgia on [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1733944625118329)

**Comment by grgia** [comment]
Closing for https://github.com/Expensify/App/issues/53656
